mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -264,7 +264,6 @@ class BisectingKMeans @Since("2.0.0") (
     val model = copyValues(new BisectingKMeansModel(uid, parentModel).setParent(this))
     val summary = new BisectingKMeansSummary(
       model.transform(dataset), $(predictionCol), $(featuresCol), $(k))
-    model.setSummary(summary)
```

Comment:
@srowen  Using another val `m` is to keep in line with `KMeans`.
But after refering to more algos (GMM/LoR/LiR), I think both `KMeans` and `BisectingKMeans` can be changed according to your comments. Thanks.